packages/rol/adapters/thyra/src/function/ROL_ThyraProductME_Objective.hpp

```diff
@@ -79,9 +79,9 @@ class ThyraProductME_Objective : public Objective<Real> {
     Thyra::ModelEvaluatorBase::InArgs<Real> inArgs = thyra_model.createInArgs();
     Thyra::ModelEvaluatorBase::OutArgs<Real> outArgs = thyra_model.createOutArgs();
 
-
     outArgs.set_g(g_index, g);
-    for(int i=0; i<p_indices.size(); ++i)
+    const int theSize = (int) p_indices.size();
```

Comment:
Also, a better name for this variable would be `p_indices_size`.  The name `theSize` does not seem well enough attracted to the array `p_indexes`.